scripts/jointLklDM.C

```diff
@@ -167,26 +171,85 @@ void jointLklDM(TString configFileName="$GLIKESYS/rcfiles/jointLklDM.rc",Int_t s
   const Double_t* massval      = massval0+nskippedmass;
   
   // Print-out configuration info (part 2)
-  cout << "*** I/O PATH            : " << fInputDataPath << endl;
+  cout << "*** I/O PATH                 : " << fInputDataPath << endl;
   cout << "***" << endl;
-  cout << "*** LABEL               : " << label <<  endl;
-  cout << "*** PROCESS             : " << strprocess << endl;
-  cout << "*** CHANNEL             : " << channel << endl;
-  cout << "*** DATA/MC             : " << simulationlabel << endl;
+  cout << "*** LABEL                    : " << label <<  endl;
+  cout << "*** PROCESS                  : " << strprocess << endl;
+  cout << "*** CHANNEL                  : " << channel << endl;
+  Int_t nChannels = 1;
+  TString* channelval = new TString[nChannels];
+  Double_t* brval = new Double_t[nChannels];
+  if(channellist.Contains(channel))
+    {
+       channelval[0] = channel;
+       brval[0] = 1.0;
+    }
+  else
+    {
+      TObjArray *coefficients = strchannel.Tokenize("+");
+      nChannels = coefficients->GetEntries();
+      channelval = new TString[nChannels];
+      brval = new Double_t[nChannels];
+      TString factorstring, coefficientstring;
+      for (Int_t iChannel = 0; iChannel < nChannels; iChannel++) 
+        {
+          factorstring = (TString)((TObjString *)(coefficients->At(iChannel)))->String();
+          TObjArray *factors = factorstring.Tokenize("*");
+          if (factors->GetEntries() != 2)
```

Comment:
You should allow for number of entries to be 1, in case BR=1, this will make (e.g.) the string "bb" be valid, so that you do not need a separate code for the nChannels=1 case

Reply:
New commit does!